File: Backtracking/javascript/1759.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");

const [l, c] = input[0].split(" ").map(Number);
const alpha = input[1].split(" ").sort();
const moSet = new Set(["a", "e", "i", "o", "u"]);
let answer = "";
const search = (s, idx, moCnt, jaCnt, len) => {
  if (len === l) {
    if (moCnt >= 1 && jaCnt >= 2) {
      answer += s + "\n";
    }
    return;
  }
  for (let i = idx; i < alpha.length; i++) {
    if (moSet.has(alpha[i])) {
      search(s + alpha[i], i + 1, moCnt + 1, jaCnt, len + 1);
      continue;
    }
    search(s + alpha[i], i + 1, moCnt, jaCnt + 1, len + 1);
  }
};

search("", 0, 0, 0, 0);
console.log(answer);

File: Backtracking/python/1759.py
from itertools import combinations
l,c = map(int,input().split())
alpha = sorted(input().split())

answer = ""
mo_set = {"a","e","i","o","u"}
def search(s):
  mo_cnt = 0
  ja_cnt = 0
  for char in s:
    if char in mo_set:
      mo_cnt+=1
    else:
      ja_cnt+=1
  if mo_cnt>=1  and ja_cnt>=2:
    return s
  return False
for s in list(combinations(alpha,l)):
  s =search("".join(s))
  if (s):
    answer += s + "\n"
print(answer)